### PR TITLE
Code for Dataflow preprocessing in census

### PR DESCRIPTION
**Pull request by fprost**
Following to this issue: https://github.com/GoogleCloudPlatform/cloudml-samples/issues/315

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/cloudml-samples/352)

### CODE REVIEW

**Comment by fprost** [comment]
For review: @gogasca

File: census/README.md [hunk 1]
```diff
@@ -388,6 +388,62 @@ gcloud ml-engine jobs describe $JOB_NAME
 Once the job is `SUCCEEDED` you can check the results in `--output-path`.
 
 
+# (Optional) Preprocessing with Dataflow
+
+**Note: This is available only for estimator/.**
```
Comment (gogasca) [inline census/README.md]: nit: Change case to Estimator + Include Link https://github.com/GoogleCloudPlatform/cloudml-samples/blob/d8eac235c64462fedde9230a6941e09b09b82ec9/census/estimator

File: census/README.md [hunk 2]
```diff
@@ -388,6 +388,62 @@ gcloud ml-engine jobs describe $JOB_NAME
...
+
+
+* **Use the updated train and eval files**
+
+```
+export TRAIN_FILE=${DATAFLOW_DIR}/output_data/train*.csv
+export EVAL_FILE=${DATAFLOW_DIR}/output_data/eval*.csv
+```
+
 ## References
```
Comment (gogasca) [inline census/README.md]: nit: After running preprocessing we can run the regular commands?

File: census/estimator/dataflow_setup.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''Dependencies for Dataflow workers.'''
+
+from setuptools import setup, find_packages
+import os
+
+NAME = 'preprocessing'
+VERSION = '1.0'
+REQUIRED_PACKAGES = ['tensorflow-transform==0.9.0']
```
Comment (gogasca) [inline census/estimator/dataflow_setup.py]: nit: CMLE supports now TF 1.12. 
By looking at the TFT matrix, there is a newer version: 0.11.0, does it makes sense to use this one?

File: census/estimator/preprocessing/preprocess.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  def read_records(self, file_name, unused_range_tracker):
+    self._file = self.open_file(file_name)
+    reader = csv.reader(self._file)
+    for rec in reader:
+      res = {key: val for key, val in zip(self._column_names, rec)}
+      yield res
+
+
+def split_data(examples, train_fraction):
```
Comment (gogasca) [inline census/estimator/preprocessing/preprocess.py]: nit: Args information, 
examples is it a pandas.Dataframe, a file? 
train_fraction is it a float ?

File: census/estimator/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+google-cloud-dataflow
+apache-beam[gcp]
+tensorflow==1.10.0
```
Comment (gogasca) [inline census/estimator/requirements.txt]: CMLE supports TF 1.12 does this be a better version than 1.10?

File: census/README.md (hunk 1 above)
Comment (fprost) [reply census/README.md]: Done

File: census/README.md (hunk 2 above)
Comment (fprost) [reply census/README.md]: Yes (make sure to not overwrite TRAIN_FILE and EVAL_FILE). 

Should I add a comment to make it more clear?

File: census/estimator/dataflow_setup.py (hunk 3 above)
Comment (fprost) [reply census/estimator/dataflow_setup.py]: Done.

File: census/estimator/requirements.txt (hunk 5 above)
Comment (fprost) [reply census/estimator/requirements.txt]: Done

File: census/estimator/preprocessing/preprocess.py (hunk 4 above)
Comment (fprost) [reply census/estimator/preprocessing/preprocess.py]: Added comment.

**Comment by fprost** [comment]
Yes ready to merge!

**Pull request merged**